vignettes/articles/sliding.Rmd

```diff
@@ -288,78 +287,151 @@ available in epipredict.
-  if (version_aware) {
-    epix_slide(
-      x,
-      ~ arx_forecaster(
-        .x, "case_rate", c("case_rate", "percent_cli"),
-        args_list = arx_args_list(ahead = ahead)
-      ) %>%
-        extract2("predictions"),
-      before = 120 - 1,
-      ref_time_values = fc_time_values,
-      new_col_name = "fc"
+have been available in real-time) to forecast the 7 day average of future
+COVID-19 case rates from current and past COVID-19 case rates and death rates
+for all states. That is, we can make forecasts on the archive, `x`, and compare
+those to forecasts on the latest data, `x_latest` using the same general set-up
+as above. For version-aware forecasting, note that `x` is fed into
+`epix_slide()`, while for version-unaware forecasting, `x_latest` is fed into
+`epi_slide()`. Note that in this example, we use a geo-pooled approach (using
+combined data from all US states and territories) to train our model.
+
+```{r load-data}
+# loading in the data
+states <- "*"
+
+confirmed_incidence_prop <- pub_covidcast(
+  source = "jhu-csse",
+  signals = "confirmed_incidence_prop",
+  time_type = "day",
+  geo_type = "state",
+  time_values = epirange(20200301, 20211231),
+  geo_values = states,
+  issues = epirange(20000101, 20211231)
+) %>%
+  select(geo_value, time_value, version = issue, case_rate = value) %>%
+  arrange(geo_value, time_value) %>%
+  as_epi_archive(compactify = FALSE)
+
+deaths_incidence_prop <- pub_covidcast(
+  source = "jhu-csse",
+  signals = "deaths_incidence_prop",
+  time_type = "day",
+  geo_type = "state",
+  time_values = epirange(20200301, 20211231),
+  geo_values = states,
+  issues = epirange(20000101, 20211231)
+) %>%
+  select(geo_value, time_value, version = issue, death_rate = value) %>%
+  arrange(geo_value, time_value) %>%
+  as_epi_archive(compactify = FALSE)
+
+
+fc_time_values <- seq(
+  from = as.Date("2020-09-01"),
+  to = as.Date("2021-12-31"),
+  by = "1 month"
+)
+
+x <- epix_merge(confirmed_incidence_prop, deaths_incidence_prop,
+  sync = "locf"
+)
+
+x <- x %>%
+  epix_slide(
+    before = 365000L, ref_time_values = fc_time_values,
+    function(x, gk, rtv) {
+      x %>%
+        group_by(geo_value) %>%
+        epi_slide_mean(case_rate, before = 6L) %>%
+        rename(case_rate_7d_av = slide_value_case_rate) %>%
+        epi_slide_mean(death_rate, before = 6L) %>%
+        ungroup() %>%
+        rename(death_rate_7d_av = slide_value_death_rate)
+    }
+  ) %>%
+  rename(version = time_value) %>%
+  rename(
+    time_value = slide_value_time_value,
+    geo_value = slide_value_geo_value,
+    case_rate = slide_value_case_rate,
+    death_rate = slide_value_death_rate,
+    case_rate_7d_av = slide_value_case_rate_7d_av,
+    death_rate_7d_av = slide_value_death_rate_7d_av
+  ) %>%
+  as_epi_archive(compactify = TRUE)
+```
+
+Here we specify the ARX model.
+
+```{r make-arx-model}
+lags <- c(0, 7, 14, 21)
+aheads <- c(7, 14, 21)
+engine <- quantile_reg()
```

Comment:
is it worth having these specified here and then input below? Maybe just specify them directly on lines 381 and 382?